src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/Delegation/DelegatedCompletionListProvider.cs

```diff
@@ -112,6 +118,24 @@ internal class DelegatedCompletionListProvider
         return rewrittenResponse;
     }
 
+    private async Task<bool> ShouldIncludeSnippetsAsync(VersionedDocumentContext documentContext, int absoluteIndex, CancellationToken cancellationToken)
+    {
+        var codeDocument = await documentContext.GetCodeDocumentAsync(cancellationToken).ConfigureAwait(false);
+        var tree = codeDocument.GetSyntaxTree();
+
+        var token = tree.Root.FindToken(absoluteIndex, includeWhitespace: true);
+
+        if (token.Kind is SyntaxKind.OpenAngle or SyntaxKind.CloseAngle)
+        {
+            return false;
+        }
+
+        var node = token.Parent;
+        var startOrEndTag = node?.FirstAncestorOrSelf<SyntaxNode>(n => RazorSyntaxFacts.IsAnyStartTag(n) || RazorSyntaxFacts.IsAnyEndTag(n));
```

Comment:
Also check that `absoluteIndex` is within the `startOrEndTag.Span`? In fact, maybe doing that can mean removing the check above?

Reply:
The above check was because if we're in `</SomeComponent $$>` there was no ancestor that was the end tag, it seems to favor as if the `>` belongs to some other MarkupText (not at my work pc right now).

That said, checking the span is a great idea to validate.